src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ImportController.php

```diff
@@ -236,6 +273,89 @@ public function downloadAction(Request $request)
         return $this->redirectToRoute('admin_import');
     }
 
+    /**
+     * Save import data match configuration.
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", message="You do not have permission to update this.", redirectRoute="admin_import")
+     * @DemoRestricted(redirectRoute="admin_import")
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function saveImportMatchAction(Request $request)
+    {
+        $formHandler = $this->get('prestashop.admin.import_data_configuration.form_handler');
+        $form = $formHandler->getForm();
+        $form->setData([
+            'match_name' => $request->request->get('match_name'),
+            'skip' => $request->request->get('skip'),
+            'type_value' => $request->request->get('type_value'),
+        ]);
+
+        $errors = $formHandler->save($form->getData());
+        $matches = [];
+
+        if (!$errors) {
+            $importMatchRepository = $this->get('prestashop.core.admin.import_match.repository');
+            $matches = $importMatchRepository->findAll();
+        }
+
+        return $this->json([
+            'errors' => $errors,
+            'matches' => $matches,
+        ]);
+    }
+
+    /**
+     * Delete import data match configuration.
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", message="You do not have permission to update this.", redirectRoute="admin_import")
+     * @DemoRestricted(redirectRoute="admin_import")
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function deleteImportMatchAction(Request $request)
+    {
+        $importMatchRepository = $this->get('prestashop.core.admin.import_match.repository');
+        $importMatchRepository->deleteById($request->get('import_match_id'));
```

Comment:
What if `import_match_id` does not refer to an existing match ?

Reply:
same as above, nothing would happen.